dhall/src/Dhall/Import.hs

```diff
@@ -774,16 +807,32 @@ getCacheFile cacheName hash = do
 
                     liftIO (Directory.setPermissions directory private)
 
-    cacheDirectory <- getCacheDirectory
+    cacheBaseDirectory <- getCacheBaseDirectory
 
-    assertDirectory (cacheDirectory </> cacheName)
+    let directory = cacheBaseDirectory </> cacheName
+    
+    let handler (_ :: IOException) = empty
 
-    let cacheFile = (cacheDirectory </> cacheName) </> ("1220" <> show hash)
+    catch (assertDirectory directory) handler
 
-    return cacheFile
+    return directory
+
+getCacheBaseDirectoryOrWarn :: (Alternative m, MonadIO m) => m FilePath
+getCacheBaseDirectoryOrWarn = do
+    let message =
+            "\n" 
+         <> "\ESC[1;33mWarning\ESC[0m: " 
+         <> "It hasn't been possible get a cache base directory from environment.\n"
+         <> "\n"
+         <> "You can provide a cache base directory by pointing the $XDG_CACHE_HOME "
```

Comment:
```suggestion
         <> "You can provide a cache base directory by pointing the $XDG_CACHE_HOME\n"
```

Reply:
Mmm, i removed them cause they already went out  the 80 default col width of the msys console and i didnt like the aspect making the lines shorter.
But it is matter of taste...

Reply:
Ah, never mind then!

Reply:
Anyway i want to paste some screenshots to show how looks